### DISCUSSION TITLE
How to make position:sticky work?

### DISCUSSION THREAD

**Discussion by rica-carv** [e107inc/e107#5321] [Q&A Themes] [answered]
I'm trying to use the bootstrap `sticky-top` class to work on a two column side menu, but it won't work, not even forcing the `position: sticky` attribute in theme css.

I want to do something similar like the item1 in this [example](https://codepen.io/ncerminara/pen/VdLpzd?editors=1100), don't need the remaining stacking of the example...

Does anyone tried that anytime and had the same issue? Is there a work around to make it work? Or i have no option to make it sticky?

**Comment by tgtje** [comment]
@rica-carv not tested, but [this explanation](https://blog.hubspot.com/website/css-position-sticky) makes more sense what happens ( i guess), in regards of the earlier mentioned.

**Reply by rica-carv** [reply]
@tgtje I tryed all the ideias on that page, but none works....

**Reply by tgtje** [reply]
well bs 4 and 3 or 5 are somewhat different. example : some news related work only for bs 5 etc..(comment made in code).
Now accepting as such it would suggest that css for the 3 versions might differ ??. (does 4 use diff styling??).
In layouts between 3 and 5 ( have no real 4) styling is also different (see layouts) .
In 5 it is coded as case with setstyle call, in 3 it is html styling. Furthermore just by reading i wonder if that position has to be made on the menu itself, or area or container, as (old way thinking : theme php splits of sorts, where div creates positioning eg.. 1 content and splits to right (or left side).  Haven't tested it, just tinkering. haven't looked into theming a long time. 
Just hopes it triggers an idea, as it is to time consuming to hunt for the differences (if there are).

**Comment by Jimmi08** [comment, marked as answer]
I just tested it on my site - boostrap 5 
It works normally  or I don't understand where problem is. It has to be used on block element (div)
I changed top value to stick it under header.
 
![image](https://github.com/user-attachments/assets/5733ec3a-6a19-4fae-95b3-4ed03e3abd6d)

**Reply by rica-carv** [reply]
@Jimmi08 It "should" normally work, but with me, it doesn't work the way i want.

Here's my page html structure
![image](https://github.com/user-attachments/assets/0c31c27c-e142-4b75-b189-3c3aedd2ec7b)

Only the whole menus area sticky-top classes work (the one highlighted). I tried to stick the single menu outer div, inside the menu area, nothing happens, and the single menu inner div, also doesn't work..., So i wonder, in order to work, i can only stick a whole menu area, not single menus inside any menu area, like i want to do.... Imagine i want to stick a single menu in area 1, another in area 2, etc....
